cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -629,13 +674,25 @@ func (m *stateMachine) setUnlockStateWithFallbackKey(partName string, unlockRes
 		} else {
 			// if we don't have an encrypted device and err != nil, then the
 			// device must be not-found, see above checks
-			m.degradedState.LogErrorf("cannot locate %s partition: %v", partName, err)
+			// it's valid to not need ubuntu-save with unencrypted partitions,
+			// so we can not report an error for ubuntu-save not being found if
+			// not encrypted, but we must then also set it as absent-but-optional
+			if partName != "ubuntu-save" {
+				m.degradedState.LogErrorf("cannot locate %s partition: %v", partName, err)
+			} else {
+				// TODO: we really need to be able to inspect the returned error
+				// from secboot.UnlockVolumeUsingSealedKeyIfEncrypted to deduce this
+				// probably
```

Comment:
I don't understand this comment, isn't it a matter of double checking unlockRes.Device ?

Reply:
yes, though now obviously we doublc check unlockRes.PartDevice